### PR TITLE
Enable CODEOWNERS

### PR DESCRIPTION
**Pull request by bloxgate**
Enables automatic review requests by file type.

@HaloSpaceStation/developers will be autorequested for code changes, and @HaloSpaceStation/mappers will be requested for any map changes. @HaloSpaceStation/administrators will be requested for any changes.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Require administrator approval in general
+*    @HaloSpaceStation/administrators
+
+# Code changes should have developer review
+*.dm @HaloSpaceStation/developers
+
+# Map changes shoudl require mapper review
+*.dmm @HaloSpaceStation/mappers
```
Comment (caelaislinn) [inline .github/CODEOWNERS:8]: Does this mean .dmm changes have to be approved by members of @HaloSpaceStation/mappers ?

**Review by caelaislinn — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (bloxgate) [reply .github/CODEOWNERS:8]: No, anyone with admin perms on the repository should be able to merge anyway if I understood the documentation right.

However that's assuming we'd even make it required in the first place:
> When someone with admin or owner permissions has enabled required reviews, they also can optionally require approval from a code owner before the author can merge a pull request in the repository. For more information, see "Enabling required reviews for pull requests."
- https://help.github.com/en/articles/about-code-owners

At the moment that setting is off.

**Comment by caelaislinn** [comment]
I don't think it's worth stratifying our team into this. We're all generalists so there actually aren't any members on the mapping team... and if there were it would be an unreasonable burden to expect them to review all mapping changes. So the end effect of this would be a lot of warning spam that gets ignored

**Comment by bloxgate** [comment]
Again, the warnings aren't enabled as mentioned in the quote in my last comment. We'd have to turn on that feature, if we wanted it. At the moment it only autorequests.

**Comment by caelaislinn** [comment]
Autorequests reviews? That's fine

**Review by caelaislinn — APPROVED** [review]

**Pull request merged**